### ISSUE TITLE
Gruppierte Dimmer hängen home.app auf

### ISSUE THREAD

**Issue by udochrist** [rdmtc/RedMatic-HomeKit#135]
Ich nutze 3 Dimmer für eine raumbeleuchtung (hm-lc-dim1l-cv). Wenn ich die Dimmer einzeln ansteuere funktioniert das in der Home App gut. Sobald ich aber die Dimmer gruppiere um sie gemeinsam zu dimmen gibts probleme. Beim einstellen des Levels gehen die Lichter korrekt an, der Knopf in der App geht allerdings dann auf Kreisel und „keine Antwort“, bis dann das ausrufungszeichen kommt. Ich hatte das früher mal mit mit ner anderen Bridge, da gabs das Problem nicht, daher glaube ich nicht dass es an der Home App, bzw. HomeKit selbst liegt.
Logfiles oder Tests jederzeit gerne.

**Comment by hobbyquaker** [comment]
An dem Thema war ich schonmal dran, da bleiben Bestätigungen von Aktoren aus, eine echte Lösung gibts da befürchte ich nicht, nur eine Kompromissfindung. Im Config Node "ccu-connection" gibts ein Setting `Queue Pause (ms)` das default auf 250ms steht. Versuch das mal (würde vorschlagen in 50ms Schritten) zu erhöhen und schau ob sich die Situation dadurch bessert (Full-Deploy oder Flow-Restart nach Änderung notwendig). 

Ein Log auf Level "Debug" schadet auch nicht, ist nicht 100% auszuschließen dass da vll noch ein anderer Bug schlummert.

**Comment by udochrist** [comment]
Das Problem ist vermutlich dass bei drei Dimmern in der Gruppe jeder Dimmer während der Rampe konstant werte liefert. Was ich nicht ganz verstanden habe: wenn der Wert des Dimmers im code gesetzt wird scheint das über ein setTimeout() zu laufen. Falls ich das richtig verstanden habe verstehe ich nicht warum das verzögert passiert.

Ich habe das gleiche Szenario nochmal mit einer anderen Bridge probiert. Dort hängt sich die App  nicht auf. Allerdings dreht der Button durch. Dadurch dass die Dimmer dauernd andere Werte liefern die gesetzt werden gibts anscheinend wieder Aktionen auf die jeweils anderen Dimmer. Schwingkreis in Software quasi. Vielleicht sollten die Updates mit value ungleich targetvalue solange ignoriert werden bis value = targetvalue ist. Evtl lohnt mal ein Blick in die impl dort für ne Idee.

**Comment by hobbyquaker** [comment]
Hast Du mal versucht die `Queue Pause` Einstellung zu erhöhen? Wie sieht es mit dem Log aus? Ist ehrlich gesagt etwas verwunderlich wenn ich konkrete Vorschläge liefere und mich dann das Gefühl beschleicht die wären nicht mal gelesen worden ;)

**Comment by udochrist** [comment]
Genau. Im code habe ich nur nachgesehen weil ich gerade nichts besseres zu tun hatte....

Ich habe den wert bis 450ms in 50er schritten erhöht. Es ändert nix am Verhalten. Was da ein sinnvoller wert sein soll ist mir ehrlich gesagt auch unklar. Wenn ich die Dimmer (Gruppe von 3en) auf einen Wert setze (meist um die 70% in den Tests) dann sind die Werte laut eq3 gui danach immer um die 30% während sich die App auf der Gruppe mit „keine Antwort“  Auch logfiles dazu habe ich, die möchte ich aber ehrlich gesagt nicht hier öffentlich einhängen. Keine Ahnung welche Infos im debug Modus da so alle auszulesen sind die niemanden was angehen.

**Comment by hobbyquaker** [comment]
Kann das Problem nachstellen, bin aber leider noch etwas ratlos. Muss da noch weiter "forschen".

**Comment by hobbyquaker** [comment]
RedMatic-HomeKit **4.0.0** ist seit heute veröffentlicht – Installation in RedMatic 9 über den Node-RED-Palettenmanager (Suche nach `redmatic-homekit`), Release-Notes: https://github.com/rdmtc/RedMatic-HomeKit/releases/tag/v4.0.0.

Dimmer werden in 4.0.0 vom Sender-Kanal gelesen und der Zustand pro Characteristic gemeldet (#319, #312) – gruppierte Dimmer sollten damit nicht mehr hängen. Das Issue ist alt und bezieht sich auf 2.x/3.x; wir schließen es deshalb erst einmal. Falls es mit 4.0.0 nicht wie erwartet funktioniert: bitte hier melden (Gerätetyp, Node-RED-Log mit Level „debug“ rund um den Start), dann wird das Issue wieder geöffnet.

---
*Written by Claude Fable on behalf of hobbyquaker.*
